pkg/config/nodetreemodel/config.go

```diff
@@ -215,6 +215,10 @@ func (c *ntmConfig) Set(key string, newValue interface{}, source model.Source) {
 
 // SetWithoutSource assigns the value to the given key using source Unknown
 func (c *ntmConfig) SetWithoutSource(key string, value interface{}) {
+	if value == nil {
+		c.UnsetForSource(key, model.SourceUnknown)
+		return
+	}
```

Comment:
Do we know where this is used ? Could is be possible to migrate use cases to UnsetForSource instead ?

Reply:
I was initially concerned with having feature drift, but then again the method `SetWithoutSource` should only be used in tests, and there only a small number of tests that rely on this behavior of how `AllSettings` acts. I updated the tests.